larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -590,19 +593,138 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
             }
         }
 
-        for (const ParticleFlowObject *const pPfoToDelete : pfoVector)
+        for (const ParticleFlowObject *const pPfoToDelete : reducedPfoVector)
         {
-            if (pPfoToEnlarge == pPfoToDelete)
+            if (selectedPfoToEnlarge == pPfoToDelete)
                 continue;
 
-            pAlgorithm->StitchPfos(pPfoToEnlarge, pPfoToDelete, pfoToLArTPCMap);
-	    stitchedPfosToX0Map.insert(PfoToFloatMap::value_type(pPfoToEnlarge, x0));
+            pAlgorithm->StitchPfos(selectedPfoToEnlarge, pPfoToDelete, pfoToLArTPCMap);
+            stitchedPfosToX0Map.insert(PfoToFloatMap::value_type(selectedPfoToEnlarge, x0));
         }
     }
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+const ParticleFlowObject *StitchingCosmicRayMergingTool::ReduceToLongestStitch(PfoVector &pfoVector, PfoVector &reducedPfoVector,
+    const ParticleFlowObject *const pPfoToEnlarge, const PfoToLArTPCMap &pfoToLArTPCMap) const
+{
+    if (pfoVector.size() > 2 )
+    {
+        reducedPfoVector = this->SelectLongestStitch(pfoVector, pfoToLArTPCMap);
+        std::sort(reducedPfoVector.begin(), reducedPfoVector.end(), LArPfoHelper::SortByNHits);
+
+        return this->GetClosestPfo(pPfoToEnlarge, reducedPfoVector);
+    }
+    else
+    {
+        for (const ParticleFlowObject *const pPfoToShift : pfoVector) reducedPfoVector.push_back(pPfoToShift);
+        std::sort(reducedPfoVector.begin(), reducedPfoVector.end(), LArPfoHelper::SortByNHits);
+
+        return pPfoToEnlarge;
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
```

Comment:
In Pandora style guide, `SelectLongestStitch` would typically have its implementation before `GetClosestPfo`, because it's called first in `ReduceToLongestStitch`.

Reply:
yeah, it was easier for me to write it in this way, but forgot to move it afterwards 

PS: Do that style guide actually exist in written form? would be useful to have in documentation...

Reply:
Bottom of slide 16 [here](https://indico.fnal.gov/event/11857/session/11/material/1/0.pdf) is as close as we've come to writing it down formally!